### PR TITLE
OS X compatibility

### PR DESCRIPTION
**Pull request by MaxLeiter**

### CODE REVIEW

File: src/state.c [hunk 1]
```diff
@@ -33,7 +38,20 @@ void set_status(struct account_state *account, enum account_status state,
...
+	mach_timespec_t mts;
+	host_get_clock_service(mach_host_self(), CALENDAR_CLOCK, &cclock);
+	clock_get_time(cclock, &mts);
+	mach_port_deallocate(mach_task_self(), cclock);
+	account->status.since.tv_sec = mts.tv_sec;
+	account->status.since.tv_nsec = mts.tv_nsec;
+	
+	#else
 	clock_gettime(CLOCK_MONOTONIC, &account->status.since);
+	#endif
```
Comment (SirCmpwn) [inline src/state.c]: Please move these into a seperate function that just returns the time in nanoseconds or something.

**Comment by SirCmpwn** [comment]
Bump

**Comment by MaxLeiter** [comment]
Should be ready (hopefully)

File: src/util/time.c [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#include "util/time.h"
+
+#ifdef __MACH__
+#include <mach/clock.h>
+#include <mach/mach.h>
+#endif
+
+void get_nanoseconds(struct timespec tp) {
```
Comment (SirCmpwn) [inline src/util/time.c]: This seems... to pass the timespec on the stack? Which would mean none of these functions do anything to it.

**Pull request merged**